### ISSUE TITLE
Errors at edge of AOI

### ISSUE THREAD

**Issue by smit1678** [hotosm/osma-health#79]
At the edge of an AOI, some OSM data is not being shown in the OSM edit recency layer. These areas are also shown as poor in the completeness layer (unclear if they are related). Also unclear if these issues are coming from OSM QA tiles (I haven't investigated that yet). 

<img width="1149" alt="openstreetmap analytics for health 2018-05-19 16-54-54" src="https://user-images.githubusercontent.com/796838/40269958-a907f1e2-5b85-11e8-9e5f-f146fccc63c8.png">

cc @geohacker @kamicut

**Comment by geohacker** [comment]
@smit1678 yes these would be tiles that are skipped because they are outside the bbox, we probably need to add a buffer but that would mean we might overestimate by a small margin.

**Comment by smit1678** [comment]
Is there a reason we’re limited to bbox and not exact/simplified shape?

Nate Smith
@nas_smith
________________________________

**Comment by geohacker** [comment]
Yes, because tile reduce uses a bbox to filter.

________________________________

**Comment by geohacker** [comment]
Agree that we may end up overestimating or underestimating. One way I can think of is using a buffer for the bbox in tile-reduce and then using the GeoJSON boundary during the analysis to throw out features that don't fall within.

**Comment by geohacker** [comment]
Oh actually, the above solution won't work in this case because we extract all buildings from a given OSM QA Tiles of a country, see [here](https://github.com/hotosm/osma-health-workers/blob/master/run.sh#L15) and [here](https://github.com/hotosm/osma-health-workers/blob/master/workers/temporal.sh#L10)

This looks like a bigger QA tiles problem to me @smit1678.

**Comment by geohacker** [comment]
I got started on switching to planet QA tiles but that does cause huge over estimation. The only way to fix it would be:
1. using planet qa tiles
2. checking each building if it falls within the boundary of the country before the analysis

The work in [progress PR is here](https://github.com/hotosm/osma-health-workers/pull/23) but this is not very straightforward and will need a lot more time. Using the boundary inside osmlint will make the process slower by a huge margin. 

The alternative approach would be stream all the buildings from osmlint through a script that checks for containment/membership and streams out the ones that are positive and then use that for analysis - this is doable, but not immediately.
